src/sql/parts/notebook/models/notebookContexts.ts

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { nb } from 'sqlops';
+
+import { localize } from 'vs/nls';
+import { IConnectionProfile } from 'sql/parts/connection/common/interfaces';
+import { IDefaultConnection, notebookConstants, INotebookModelOptions } from 'sql/parts/notebook/models/modelInterfaces';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { IConnectionManagementService } from 'sql/parts/connection/common/connectionManagement';
+import { ConnectionProfile } from 'sql/parts/connection/common/connectionProfile';
+
+export class NotebookContexts {
+	private static MSSQL_PROVIDER = 'MSSQL';
+
+	private static get DefaultContext(): IDefaultConnection {
+		let defaultConnection: ConnectionProfile = <any>{
+			providerName: NotebookContexts.MSSQL_PROVIDER,
+			id: '-1',
+			serverName: localize('selectConnection', 'Select connection')
+		};
+
+		return {
+			// default context if no other contexts are applicable
+			defaultConnection: defaultConnection,
+			otherConnections: [defaultConnection]
+		};
+	}
+
+	private static get LocalContext(): IDefaultConnection {
+		let localConnection: ConnectionProfile = <any>{
+			providerName: NotebookContexts.MSSQL_PROVIDER,
+			id: '-1',
+			serverName: localize('localhost', 'Localhost')
```

Comment:
I think `localhost` lowercased is a valid connection (e.g. server called localhost). I think "Not Connected" would be more accurate??

Reply:
At least in the SQL case... I know that for PySpark etc. it might be different. We should ask others about this.

Reply:
Heh, so... Fun fact, even though localConnection's provider is MSSQL, this code can only possibly be hit when a connection provider id isn't provided by a kernel.

The reason why MSSQL is the provider name is that I needed to put... _something_ there, and didn't want any Jupyter references, so that's the story there.